### PR TITLE
Vite integration

### PR DESCRIPTION
**Pull request by lukicenturi**
## Checklist

- [x] Vite for serving web
- [x] Vite for serving electron
- [x] Vite for building web
- [x] Vite for building electron
- [x] Integrate e2e testing (cypress)
- [x] Research vitest to replace jest as unit tester
- [x] Finalize removing all vue-cli-service-usage

image

### CODE REVIEW

**Comment by kelsos** [comment]
So excited about this :fireworks:

**Comment by kelsos** [comment]
@lukicenturi when you are done with everything else make also sure that vite is at [2.9.0](https://twitter.com/vite_js/status/1509165446595072001?s=20&t=WQeQT6j1YIqVDr0--WuzOQ)

It seems to have further speed improvmements

**Comment by kelsos** [comment]
Now that the release is done we need to rebase again on top of develop
And see what else is missing to merge it

File: frontend/app/package.json [hunk 1]
```diff
@@ -4,45 +4,46 @@
...
+    "i18n:report": "vue-i18n-extract report --vueFiles './src/**/*.?(ts|js|vue)' --languageFiles './src/locales/**/*.json'",
+    "lint:check": "eslint ./ --no-fix --max-warnings 0  --ext .js,.ts,.vue,.json",
     "preinstall": "node ../check-versions.js",
     "postinstall": "electron-builder install-app-deps",
     "postuninstall": "electron-builder install-app-deps",
     "serve:backend": "node start-backend.js",
     "lint:style": "stylelint 'src/**/*.{vue,scss}'",
     "lint:style:fix": "stylelint --fix 'src/**/*.{vue,scss}'",
-    "test:e2e-ci": "nyc vue-cli-service test:e2e --mode e2e --headless",
+    "test:e2e-ci": "dotenv -e .env.e2e start-server-and-test serve http-get://localhost:8080 cypress:run",
```
Comment (kelsos) [inline frontend/app/package.json:24]: Do you think we can get the coverage working with cypress now that we got rid of webpack?

File: frontend/tsconfig.json [hunk 2]
```diff
@@ -15,8 +15,8 @@
     "baseUrl": ".",
     "types": [
       "vuetify",
-      "webpack",
-      "webpack-env"
+      "vite/client",
+      "unplugin-vue2-script-setup/types"
```
Comment (kelsos) [inline frontend/tsconfig.json:19]: can we start migrating to `<script setup>` :smile:

File: frontend/app/src/components/staking/kraken/KrakenStakingOverview.vue [hunk 3]
```diff
@@ -87,7 +87,7 @@ export default defineComponent({
   name: 'KrakenStakingOverview',
   components: { ValueAccuracyHint },
   props: {
-    totalUsd: { required: true, type: Object as PropType<BigNumber> },
+    totalUsd: { required: true, type: BigNumber },
```
Comment (kelsos) [inline frontend/app/src/components/staking/kraken/KrakenStakingOverview.vue:90]: doesn't this give problems?

Comment (lukicenturi) [reply frontend/app/src/components/staking/kraken/KrakenStakingOverview.vue:90]: `type: BigNumber` works well. 
The old implementation throws warning messages.

File: frontend/tsconfig.json (hunk 2 above)
Comment (lukicenturi) [reply frontend/tsconfig.json:19]: yes we can 😄

Comment (LefterisJP) [reply frontend/tsconfig.json:19]: I am sorry for barging in but I just had to

yes-we-can-yaby2n

**Comment by kelsos** [comment]
Hmm ok the e2e tests situation is tricky. This failure doesn't happen locally, and the videos seem to freeze so there is not much I could get out of them.

Since the subsequent tests seem to pass the problem seems to be with loading the first one @lukicenturi any ideas?

File: frontend/app/package.json (hunk 1 above)
Comment (kelsos) [reply frontend/app/package.json:24]: I created an issue for this #4391

**Comment by kelsos** [comment]
@lukicenturi one thing I noticed is that ARGS are not passed properly now to `cypress run`

**Comment by kelsos** [comment]
btw check if we can also make use https://www.npmjs.com/package/start-server-and-test#starting-two-servers of this

**Comment by lukicenturi** [comment]
> Hmm ok the e2e tests situation is tricky. This failure doesn't happen locally, and the videos seem to freeze so there is not much I could get out of them.
> 
> Since the subsequent tests seem to pass the problem seems to be with loading the first one @lukicenturi any ideas?

Yeah I also noticed that. 
I'm not sure, but last time I rebase vite, I had to downgrade cypress version to make it work in CI.

**Comment by lukicenturi** [comment]
> @lukicenturi one thing I noticed is that ARGS are not passed properly now to `cypress run`

hmm yeah, I noticed I have to do something with `test:e2e-ci` since we get rid of vue-cli-service

**Comment by lukicenturi** [comment]
@kelsos voila the e2e works.
I noticed that the first app.visit() always failed in CI.
So I just add one app.visit() before running the tests.
wdyt?

**Review by kelsos — APPROVED** [review]

**Pull request merged**